src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -0,0 +1,890 @@
+    }
+}
+
+
+void
+SDCPUThread::onDataAddrTranslated(weak_ptr<InflightInst> inst, Fault fault,
+                                  const RequestPtr& req, bool write,
+                                  uint8_t* data,
+                                  shared_ptr<SplitRequest> sreq, bool high)
+{
+    const shared_ptr<InflightInst> inst_ptr = inst.lock();
+    if (!inst_ptr || inst_ptr->isSquashed()) {
+        // No need to do anything for an instruction that has been squashed.
+        return;
+    }
+
+    DPRINTF(SDCPUThreadEvent, "onDataAddrTranslated()\n");
+
+    if (fault != NoFault) {
+        markFault(inst_ptr, fault);
+
+        return;
+    }
+
+    // TODO set the effective physical address for the inflightInst
+    // TODO add dependencies according to the now known effective address, and
+    //      unset this instruction ready if appropriate.
+
+    auto callback = [this, inst](Fault fault) {
+            onExecutionCompleted(inst, fault);
+        };
+
+    // IF we have a split request
+    if (sreq) {
+        if (high) {
+            sreq->high = req;
+        } else {
+            sreq->low = req;
+        }
+
+        // IF both requests have been translated successfully
+        if (sreq->high && sreq->low) {
+            if (write) {
+                if (!_cpuPtr->requestSplitMemWrite(sreq->main, sreq->low,
+                        sreq->high, getThreadContext(), inst_ptr->staticInst(),
+                        static_pointer_cast<ExecContext>(inst_ptr), data,
+                        callback)) {
+                    panic("The CPU rejected my mem access request and I "
+                          "haven't been programmed to know how to "
+                          "continue!!!");
+                }
+            } else { // read
+                if (!_cpuPtr->requestSplitMemRead(sreq->main, sreq->low,
+                        sreq->high, getThreadContext(), inst_ptr->staticInst(),
+                        static_pointer_cast<ExecContext>(inst_ptr),
+                        callback)) {
+                    panic("The CPU rejected my mem access request and I "
+                          "haven't been programmed to know how to "
+                          "continue!!!");
+                }
+            }
+        }
+
+        return; // finished handling split request
+    }
+
+    if (write) {
+        if (!_cpuPtr->requestMemWrite(req, getThreadContext(),
+                inst_ptr->staticInst(),
+                static_pointer_cast<ExecContext>(inst_ptr), data, callback)) {
+            panic("The CPU rejected my mem access request and I haven't been "
+                  "programmed to know how to continue!!!");
+        }
+    } else { // read
+        if (!_cpuPtr->requestMemRead(req, getThreadContext(),
+                inst_ptr->staticInst(),
+                static_pointer_cast<ExecContext>(inst_ptr),
+                callback)) {
+            panic("The CPU rejected my mem access request and I haven't been "
```

Comment:
We need to think about how to handle this.

Reply:
I agree. Right now, the true-false accept-reject nature of the relationship between CPU and thread is kind of in a weird place, because the responsibility of buffering requests or stalling is not necessarily well-defined in this model yet.